mlrun/api/initial_data.py

```diff
@@ -331,6 +335,43 @@ def _find_last_updated_artifact(
     return last_updated_artifact
 
 
+def _perform_version_2_data_migrations(
+    db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    _align_runs_table(db, db_session)
+
+
+def _align_runs_table(
+    db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    logger.info("Aligning runs")
+    runs = db._find_runs(db_session, None, "*", None).all()
+    for run in runs:
+        run_dict = run.struct
+
+        # Align run start_time column to the start time from the body
+        run.start_time = (
+            mlrun.api.db.sqldb.helpers.run_start_time(run_dict) or run.start_time
+        )
+
+        # New name column added, fill it up from the body
+        run.name = run_dict.get("metadata", {}).get("name", "no-name")
+
+        # State field used to have a bug causing only the body to be updated, align the column
+        run.state = run_dict.get("status", {}).get(
+            "state", mlrun.runtimes.constants.RunStates.created
+        )
+
+        # New updated column added, fill it up from the body
+        updated = datetime.datetime.now(tz=datetime.timezone.utc)
+        if run_dict.get("status", {}).get("last_update"):
+            updated = dateutil.parser.parse(
+                run_dict.get("status", {}).get("last_update")
+            )
+        run.updated = updated
```

Comment:
```suggestion
        # New updated column added, fill it up from the body
        updated = datetime.datetime.now(tz=datetime.timezone.utc)
        db._update_run_updated_time(run, run_dict, updated)
```
Are you not using it to avoid private functions?

Reply:
I do need the if to resolve the time I should use (precedence is to use the body time, only if doesn't exists, using `now()`), but it is correct that in the case I'm using `now()` I better update the body as well